src/Loader/LoaderInterface.php

```diff
@@ -24,14 +26,16 @@ interface LoaderInterface
      *
      * @param mixed $resource Resource to load
      *
+     * @phpstan-assert-if-true LoaderInterface<TResourceType> $this
```

Comment:
this looks weird to me. This is always the type of `$this` due to the generic type.

`@phpstan-assert-if-true =TResourceType $resource` would make more sense (the `=` prefix indicates that returning `false` does not mean that the resource is not of that type, as we might have additional checks, for instance that the string corresponds to a valid file path)

Reply:
I was doing that specifically for `resolveLoader()` to work well (in PHPStan's perspective).

Using your code breaks it, but now I found out it also breaks on PHPStan level 9.

I did make it work by combining your code with mine though:

https://phpstan.org/r/862e103a-f937-4713-b339-ea212db5649d

(the key difference is the PHPDoc of `LoaderInterface::supports()`)